packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -457,6 +458,62 @@ async function doGenerateArmTemplate(
 
   // Write bicep content to project folder
   if (bicepOrchestrationTemplate.needsGenerateTemplate()) {
+    // Output parameter file
+    const envListResult = await environmentManager.listEnvConfigs(ctx.root);
+    if (envListResult.isErr()) {
+      return err(envListResult.error);
+    }
+    const parameterEnvFolderPath = path.join(ctx.root, configsFolder);
+    await fs.ensureDir(parameterEnvFolderPath);
+    for (const env of envListResult.value) {
+      const parameterFileName = parameterFileNameTemplate.replace(EnvNamePlaceholder, env);
+      const parameterEnvFilePath = path.join(parameterEnvFolderPath, parameterFileName);
+      let parameterFileContent = "";
+      if (await fs.pathExists(parameterEnvFilePath)) {
+        try {
+          const parameterFile = await fs.readJson(parameterEnvFilePath);
+          const paramterObj = parameterFile.parameters.provisionParameters.value;
+          const appendParam = bicepOrchestrationTemplate.getAppendedParameters();
+          const duplicateParam = Object.keys(paramterObj).filter((val) =>
+            Object.keys(appendParam).includes(val)
+          );
+          if (duplicateParam && duplicateParam.length > 1) {
+            const returnError = new Error(
+              `There are some duplicate parameters in ${parameterEnvFilePath}, to avoid the conflict, please modify these parameter names except resourceBaseName: ${duplicateParam}`
```

Comment:
we may not report resourceBaseName as duplicate parameter to avoid making user confuse

Reply:
sure